src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/w3c/JavascriptW3cPerformanceTimeline.kt

```diff
@@ -1,117 +1,31 @@
 package com.atlassian.performance.tools.jiraactions.api.w3c
 
+import com.atlassian.performance.tools.jiraactions.api.w3c.harvesters.getJsElementsPerformance
+import com.atlassian.performance.tools.jiraactions.api.w3c.harvesters.getJsNavigationsPerformance
+import com.atlassian.performance.tools.jiraactions.api.w3c.harvesters.getJsResourcesPerformance
 import org.openqa.selenium.JavascriptExecutor
-import java.time.Duration
 
 /**
  * Obtains entries from [javascript].
  */
 class JavascriptW3cPerformanceTimeline(
-    private val javascript: JavascriptExecutor
+    javascript: JavascriptExecutor,
+    withNavigationPerformance: Boolean = true,
+    withResourcesPerformance: Boolean = true,
+    withElementPerformance: Boolean = true
 ) : W3cPerformanceTimeline {
+    private var getNavigationPerformance: () -> List<PerformanceNavigationTiming> =
```

Comment:
`Supplier` fileds are basically functions:
```suggestion
    private fun getNavigationPerformance() : List<PerformanceNavigationTiming> =
```

Reply:
Seems like it won't work - the function implementation differs based on the `with...` field. I think it has to remain a var :/ 
Or it's just me not being a Kotlin master yet ;P

Reply:
It's because these are wrapped in too many `{ }`, turning them into lambdas.
Here's a few adjustments: https://github.com/atlassian-forks/jira-actions/pull/1

Reply:
Thanks for that!